src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -404,7 +404,32 @@ private function parseFileToDOM(string $file): \DOMDocument
         try {
             $dom = XmlUtils::loadFile($file, [$this, 'validateSchema']);
         } catch (\InvalidArgumentException $e) {
-            throw new InvalidArgumentException(sprintf('Unable to parse file "%s": ', $file).$e->getMessage(), $e->getCode(), $e);
+            $invalidSecurityElements = [];
+            $errors = explode("\n", $e->getMessage());
+            foreach ($errors as $i => $error) {
+                if (preg_match("#^\[ERROR 1871] Element '\{http://symfony\.com/schema/dic/security}([^']+)'#", $error, $matches)) {
+                    $invalidSecurityElements[$i] = $matches[1];
+                }
+            }
+            if ($invalidSecurityElements) {
+                $dom = XmlUtils::loadFile($file);
+
+                foreach ($invalidSecurityElements as $errorIndex => $tagName) {
+                    foreach ($dom->getElementsByTagName($tagName) as $element) {
+                        if (!$parent = $element->parentElement) {
+                            continue;
+                        }
+                        if ('provider' === $parent->tagName || 'firewall' === $parent->tagName) {
+                            trigger_deprecation('symfony/security-bundle', '5.4.41', 'Third-party %s must now be namespaced; please update your security configuration "%s" tag.', 'provider' === $parent->tagName ? 'providers' : 'authenticators', $tagName);
```

Comment:
let's move the deprecation to 7.2, we don't need to report it in 5.4, do we?

Reply:
The issue is related to libxml rather than Symfony so it would be best to report to as many users as possible.

Reply:
we still shouldn't add new deprecations to 5.4, they are exclusively for 7.2
since we can deal with the situation, all is fine for 5.4/6.4

Reply:
Okay, removing the call to `trigger_deprecation` :ok_hand: